### PR TITLE
[CI] Align MiniMax H3 Ref2VA inputs and add I2VA/Ref2VA accuracy coverage

### PR DESCRIPTION
**Pull request by david6666666**
## Purpose

Align MiniMax H3 Ref2VA model-card input handling and keep reproducible official-asset accuracy coverage for I2VA and Ref2VA.

- Bound reference video and audio conditions to the generated temporal window.
- Preserve RGB reference pixels losslessly through preprocessing.
- Normalize audio timestamps so muxed AAC does not expose encoder priming as leading silence.
- Expose standalone audio encoding in the MiniMax H3 profiler.
- Add SHA-pinned model-card assets and shared metadata/audio/video similarity checks for I2VA and Ref2VA.
- Use one accuracy gate for both retained cases: SSIM >= 0.97 and PSNR >= 34 dB.
- Remove the obsolete T2VA accuracy case and update the CUDA nightly path.

## Test Plan

**vLLM Version:** 0.26.0

**Hardware:** 4x NVIDIA B300, BF16, Ulysses 4, ring 1, HSDP shard size 4, VAE patch parallel 4, tiled VAE, FlashAttention, 50 steps, seed 0.

image

- `pre-commit run --files .buildkite/cuda/test-nightly.yml tests/e2e/accuracy/minimax_h3/test_minimax_h3_i2va_ref2va_similarity.py`
- `pytest --collect-only -q tests/e2e/accuracy/minimax_h3/test_minimax_h3_i2va_ref2va_similarity.py` (2 tests collected)
- Contract/API validation: 167 passed.
- I2VA: SSIM `0.975190`, PSNR `35.981286 dB`, passing the common gate.
- Ref2VA: SSIM `0.980577`, PSNR `42.650204 dB`, passing the common gate.
- Both retained cases match the official 1344x768, 24 FPS, AAC 32 kHz stereo contract; frame counts are 192 and 124 respectively.

The T2VA case was removed from the retained accuracy/nightly coverage because the current strict gate was not met; this PR does not claim T2VA pixel-level alignment.

### CODE REVIEW

**Comment by david6666666** [comment]
Validation update on commit `dec18b95`:

- 4x NVIDIA B300 (physical devices 4-7), BF16, Ulysses 4, ring 1, HSDP shard size 4, tiled VAE, FlashAttention, 50 steps, seed 0.
- T2VA: passed; SSIM `0.849060`, PSNR `21.403426 dB`.
- I2VA: passed; SSIM `0.975190`, PSNR `35.981286 dB`.
- Ref2VA: passed; SSIM `0.980567`, PSNR `42.649689 dB`.
- All three outputs matched the official 1344x768, 24 FPS, AAC 32 kHz stereo container contract.

The accuracy module now collects three model-card regression cases and pins the official output/input assets with SHA-256 checks.

**Comment by david6666666** [comment]
Validation update on commit c875d7fd:

- T2VA now sends the canonical model-card payload: task=t2va, conditions=[], target.short_edge=768, aspect_ratio=16:9, duration_seconds=10, seed=0.
- T2VA gates are now SSIM >= 0.97 and PSNR >= 34 dB; the shared helper keeps the existing I2VA/Ref2VA gates unchanged.
- 4x B300 validation on physical GPUs 4-7 completed successfully through generation and media-contract checks: 1344x768, 243 frames, 24 fps, 32 kHz audio.
- The strict similarity gate currently fails with SSIM=0.875570 and PSNR=23.398015 dB. This is intentionally recorded as a failing accuracy result; T2VA is not yet aligned to the requested threshold.
- py_compile, ruff, ruff format, git diff --check, and pre-commit passed.

**Comment by david6666666** [comment]
Validation update on commit `ef0c0835`:

- All three MiniMax H3 accuracy cases now share `SSIM_THRESHOLD = 0.97` and `PSNR_THRESHOLD = 34.0`.
- T2VA uses the canonical model-card request: `task=t2va`, empty conditions, 768 short edge, 16:9, 10 seconds, and seed 0.
- On 4x B300 (physical GPUs 4-7), the generated T2VA output passed the media contract. Against the official video: SSIM `0.847311`, PSNR `21.217689 dB`.
- The independently generated reference output against the same official video measured SSIM `0.781898`, PSNR `18.498464 dB`; generated-vs-reference measured SSIM `0.752910`, PSNR `17.741266 dB`.
- Therefore the requested strict T2VA gate is currently a real failing gate; this update does not claim T2VA accuracy alignment.
- Targeted `py_compile`, Ruff, formatting, `git diff --check`, and pre-commit checks passed.

File: vllm_omni/diffusion/models/minimax_h3/pipeline_minimax_h3.py [hunk 1]
```diff
@@ -1204,14 +1205,26 @@ def _encode_audio_condition(
...
             return None, []
         _, rank, _ = _dit_rank_world()
         rows = None
         lengths = torch.zeros(len(audios), dtype=torch.long, device=self.device)
         if rank == 0:
-            encoded = [self.audio_vae.encode_waveform(*audio) for audio in audios]
+            if max_duration_seconds is not None:
+                max_duration_seconds = float(max_duration_seconds)
+                if max_duration_seconds <= 0:
+                    raise ValueError("max_duration_seconds must be positive")
```
Comment (yuanheng-zhao) [inline vllm_omni/diffusion/models/minimax_h3/pipeline_minimax_h3.py]: This probably causes a hang (e.g., rank 0 raises, other ranks blocks at 

```python
        if world_size > 1:
            dist.broadcast(lengths, src=0, group=group)
```

Better move the check before the `rank == 0` branch

File: tests/diffusion/models/minimax_h3/test_minimax_h3_contract.py [hunk 2]
```diff
@@ -1246,8 +1246,39 @@ def test_r7_r8_ref2va_video_segment_matrix(monkeypatch, tmp_path, case, start_ti
...
+    sampled = reference_video_module.sample_reference_video_frames(
+        "prepared.mp4",
+        workdir="unused",
+    )
+
+    assert load_calls == ["prepared.mp4"]
+    assert [int(frame[0, 0, 0]) for frame in sampled["frames"]] == [
+        0,
+        int(decoded[12, 0, 0, 0]),
+        int(decoded[24, 0, 0, 0]),
```
Comment (yuanheng-zhao) [inline tests/diffusion/models/minimax_h3/test_minimax_h3_contract.py]: This test seems to be weak and test itself, rather than impl logics

File: tests/diffusion/models/minimax_h3/test_minimax_h3_contract.py [hunk 3]
```diff
@@ -1246,8 +1246,39 @@ def test_r7_r8_ref2va_video_segment_matrix(monkeypatch, tmp_path, case, start_ti
...
+    )
+    monkeypatch.setattr(
+        reference_video_module,
+        "load_video_frames",
+        lambda path: load_calls.append(path) or decoded,
+    )
+
+    sampled = reference_video_module.sample_reference_video_frames(
+        "prepared.mp4",
+        workdir="unused",
```
Comment (yuanheng-zhao) [inline tests/diffusion/models/minimax_h3/test_minimax_h3_contract.py]: nit: workdir dead arg

File: vllm_omni/diffusion/models/minimax_h3/pipeline_minimax_h3.py (hunk 1 above)
Comment (david6666666) [reply vllm_omni/diffusion/models/minimax_h3/pipeline_minimax_h3.py]: Fixed in the latest head. `max_duration_seconds` is now validated before entering the rank-0 branch, so invalid input raises on every rank before any collective. Added a regression test that exercises the non-zero-rank path.

File: tests/diffusion/models/minimax_h3/test_minimax_h3_contract.py (hunk 2 above)
Comment (david6666666) [reply tests/diffusion/models/minimax_h3/test_minimax_h3_contract.py]: Addressed. The sampling test now verifies the complete sampled frame arrays and block timestamps, while still asserting that the prepared video is decoded exactly once.

File: tests/diffusion/models/minimax_h3/test_minimax_h3_contract.py (hunk 3 above)
Comment (david6666666) [reply tests/diffusion/models/minimax_h3/test_minimax_h3_contract.py]: Addressed. Removed the unused `workdir` parameter from the sampling helper and all call sites.

**Comment by hsliuustc0106** [comment]
https://buildkite.com/vllm/vllm-omni/builds/13422/canvas?sid=019feea3-4aad-4b03-ae22-5f2ed18656fc&tab=output is this Linbot-world acc failure related or not?

**Comment by Gaohan123** [comment]
> https://buildkite.com/vllm/vllm-omni/builds/13422/canvas?sid=019feea3-4aad-4b03-ae22-5f2ed18656fc&tab=output is this Linbot-world acc failure related or not?

Also happened in the latest nightly CI: https://buildkite.com/vllm/vllm-omni/builds/13407/canvas?sid=019fecd9-14c2-476d-be9c-1f55634370e3&tab=output

**Review by Gaohan123 — APPROVED** [review]
LGTM. Thanks
